Add ingress2gateway information in Getting Started guide (#1056)
nginx/documentation: content/ngf/get-started.md

Request ID: 1b5c8f3a4532ea7d7510cf14ffeb27af
```

> ## Migrating from NGINX Ingress Controller

Reviewer: ```suggestion
## Migrate from NGINX Ingress Controller
```

Imperative sentence case: we don't use gerunds.

If you're already using NGINX Ingress Controller and want to migrate to NGINX Gateway Fabric, you can use the [ingress2gateway](https://github.com/kubernetes-sigs/ingress2gateway) tool to automatically convert your existing Ingress resources to Gateway API resources.

> The ingress2gateway tool is a Kubernetes SIG project that helps convert Ingress resources to Gateway API resources. It supports various Ingress providers, including NGINX Ingress Controller.

Reviewer: ```suggestion
The ingress2gateway tool is a Kubernetes SIG project for converting Ingress resources to Gateway API resources. It supports multiple Ingress providers, including NGINX Ingress Controller.
```

> The ingress2gateway tool is a Kubernetes SIG project that helps convert Ingress resources to Gateway API resources. It supports various Ingress controllers, including NGINX Ingress Controller.

Reviewer 2: ```suggestion
The ingress2gateway tool is a Kubernetes SIG project that helps convert Ingress resources to Gateway API resources. It supports various providers, including NGINX Ingress Controller.
```

matches the language in the tool

{{< call-out "important" >}}
The ingress2gateway tool is a conversion utility that translates Ingress resources to Gateway API equivalents. It is not a complete end-to-end migration solution. You will need to manually review the converted resources, test functionality, and make additional configuration changes as needed for your specific environment.
{{< /call-out >}}

> ### Using ingress2gateway

Reviewer: ```suggestion
### Use ingress2gateway
```

To convert your existing NGINX Ingress resources to Gateway API resources that work with NGINX Gateway Fabric:

1. Install the ingress2gateway tool following the [installation instructions](https://github.com/kubernetes-sigs/ingress2gateway?tab=readme-ov-file#installation).

2. Run the conversion command for NGINX provider:
   ```shell
   ingress2gateway print --providers=nginx --input-file=<your-ingress-file>
   ```

> 3. Review the generated Gateway API resources and apply them to your cluster

Reviewer 3: probably should mention what the conversion command does. Something along the lines of, "... this will generate an output file named ______ which contains all the generated Gateway API resources". I'd just like some information on what the command does and where the results would be.

Reviewer 2: i think adding the location information would be helpful here as well.

Reviewer 3: Hm, I see. Yea I think an output redirect would be better since I'd expect with complicated conversions there can be a lot of resources which might even flood some people's terminals max lines.

>    This command will analyze your Ingress resources and output the equivalent Gateway API resources to stdout. You can redirect the output to a file for review:

Reviewer 3: ```suggestion
   This command will analyze your Ingress resources from the input file and output the equivalent Gateway API resources to a file named `gateway-api0resources.yaml`
```

> 1. Install the ingress2gateway tool following the [installation instructions](https://github.com/kubernetes-sigs/ingress2gateway?tab=readme-ov-file#installation).
> 
> 2. Run the conversion command for NGINX provider:
>    ```shell
>    ingress2gateway print --providers=nginx --input-file=<your-ingress-file> > gateway-api-resources.yaml
>    ```
>    This command will analyze your Ingress resources from the input file and output the equivalent Gateway API resources to a file named `gateway-api-resources.yaml`
> 
> 3. Review the generated Gateway API resources in the output file and apply them to your cluster:
>    ```shell
>    kubectl apply -f gateway-api-resources.yaml
>    ```

Reviewer: ```suggestion

First, [install the ingress2gateway tool](https://github.com/kubernetes-sigs/ingress2gateway?tab=readme-ov-file#installation).

Then run the conversion command for the NGINX provider:
   
```shell
ingress2gateway print --providers=nginx --input-file=<your-ingress-file> > gateway-api-resources.yaml
```
This tool will analyze your Ingress resources from the input file and output the equivalent Gateway API resources to a file named `gateway-api-resources.yaml`

Finally, review the generated Gateway API resources in the output file and apply them to your cluster:

```shell
kubectl apply -f gateway-api-resources.yaml
```

```

Indented lists tend to break formatting in weird ways, and people read top to bottom so there is no need to use listed for nested, numbered steps.

Reviewer: GitHub has absolutely ate it trying to format this suggestion with nested codeblocks, but the intent should be mostly there. I can push a commit if it's not.

After conversion, you may need to adjust some configurations to take advantage of NGINX Gateway Fabric-specific features or to ensure compatibility with your existing setup.

> {{< /call-out >}}

Reviewer 2: seems repetitive for me, why can't we add this information as part of the first call-out ?

The ingress2gateway tool is a Kubernetes SIG project that helps convert Ingress resources to Gateway API resources. It supports various Ingress controllers, including NGINX Ingress Controller.

{{< call-out "important" >}}
The ingress2gateway tool is a conversion utility that translates Ingress resources to Gateway API equivalents. It is not a complete end-to-end migration solution. You will need to manually review the converted resources, test functionality, and make additional configuration changes as needed for your specific environment.
{{< /call-out >}}

### Using ingress2gateway

To convert your existing NGINX Ingress resources to Gateway API resources that work with NGINX Gateway Fabric:

1. Install the ingress2gateway tool following the [installation instructions](https://github.com/kubernetes-sigs/ingress2gateway).

2. Run the conversion command for NGINX provider:
   ```shell
   ingress2gateway --providers=nginx --input-file=<your-ingress-file>.yaml > output-file=gateway-api-resources.yaml

>    ```

Author: Oh it shouldn't be I think I didn't add the output file in that one and also I need to update this

>    ```

Author: oh shoot yeah good catch thanks

>    ```

Author: fixed the print part, but I left the input file like <input file> since I think that's better. I can update the ingress2gateway readme for the input file part

>    This command will analyze your Ingress resources and output the equivalent Gateway API resources to stdout. You can redirect the output to a file for review:

Reviewer 3: honestly i think we should remove the first command which prints it to stdout. I feel like it won't be as popular as generating a file. If you leave it in, we should adjust step 3 which doesn't mention the printing to stdout as it seems like it automatically assumes they've printed to the output file.
